building/build-debs/homeworld-keysystem/src/keyclient/actloop/actloop_test.go

```diff
@@ -0,0 +1,387 @@
+package actloop
```

Comment:
Some questions about how `keyclient` works in general:
- Why is so much state stored on disk, rather than in the `state` struct?
- Doesn't the keygranting cert expire?  I don't see where we renew it.

Reply:
What do you mean about "so much state" being stored on disk? Keys and certs are stored on disk, as are downloaded files, because that's where downstream consumers pull them from. The only thing I can think of that you might be thinking about would be the bootstrap token, which needs to be on disk so that it's preserved across restarts of the service.

The keygranting cert is renewed by having an entry in the `keys` section to handle renewing it. Something like this:

      - name: keygranting
        type: tls
        key: /etc/hyades/keyclient/granting.key
        cert: /etc/hyades/keyclient/granting.pem
        api: renew-keygrant
        in-advance: 336h   # renew two weeks before expiration

Reply:
I am still a little confused.  Where in the key renewal process does the `Keygrant` field get written to, so that we start using the new keygranting cert instead of the old one?

Reply:
Ah, woops -- I apparently dropped the call to ReloadKeygrantingCert from RequestOrRenewAction.Perform() during refactoring.